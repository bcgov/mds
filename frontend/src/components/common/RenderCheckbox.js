import React from "react";
import PropTypes from "prop-types";
import { Form, Checkbox } from "antd";

/**
 * @constant RenderCheckbox - Ant Design `Checkbox` component for redux-form.
 */

const propTypes = {
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  label: PropTypes.string,
  meta: PropTypes.object,
};

const RenderCheckbox = ({ id, input, label, meta: { touched, error } }) => (
  <Form.Item validateStatus={touched ? error && "error" : ""}>
    <Checkbox id={id} checked={input.value} {...input}>
      {label}
    </Checkbox>
  </Form.Item>
);

RenderCheckbox.propTypes = propTypes;

export default RenderCheckbox;
