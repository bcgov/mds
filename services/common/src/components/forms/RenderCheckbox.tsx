import React, { useState } from "react";
import PropTypes from "prop-types";
import { Form } from "@ant-design/compatible";
import "@ant-design/compatible/assets/index.css";
import { Checkbox } from "antd";
/**
 * @constant RenderCheckbox - Ant Design `Checkbox` component for redux-form.
 */

const propTypes = {
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  label: PropTypes.string.isRequired,
  meta: PropTypes.objectOf(PropTypes.any).isRequired,
  input: PropTypes.objectOf(PropTypes.string).isRequired,
  disabled: PropTypes.bool.isRequired,
};

const RenderCheckbox = (props) => {
  const [checked, setChecked] = useState(props.input.value || false);
  const onChange = (e) => {
    setChecked(e.target.checked);
  };
  return (
    <Form.Item validateStatus={props.meta.touched ? props.meta.error && "error" : ""}>
      <Checkbox
        id={props.id}
        {...props.input}
        disabled={props.disabled}
        checked={checked}
        onChange={onChange}
      >
        {props.label}
      </Checkbox>
    </Form.Item>
  );
};

RenderCheckbox.propTypes = propTypes;

export default RenderCheckbox;
