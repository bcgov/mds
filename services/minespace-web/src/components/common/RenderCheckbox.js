import React from "react";
import PropTypes from "prop-types";
import { Form } from "@ant-design/compatible";
import "@ant-design/compatible/assets/index.css";
import { Checkbox } from "antd";

/**
 * @constant RenderCheckbox - Ant Design `Checkbox` component for redux-form.
 */

const propTypes = {
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  label: PropTypes.string.isRequired,
  meta: PropTypes.objectOf(PropTypes.any).isRequired,
  input: PropTypes.objectOf(PropTypes.string).isRequired,
  disabled: PropTypes.bool.isRequired,
};

const RenderCheckbox = (props) => (
  <Form.Item validateStatus={props.meta.touched ? props.meta.error && "error" : ""}>
    <Checkbox id={props.id} checked={props.input.value} {...props.input} disabled={props.disabled}>
      {props.label}
    </Checkbox>
  </Form.Item>
);

RenderCheckbox.propTypes = propTypes;

export default RenderCheckbox;
