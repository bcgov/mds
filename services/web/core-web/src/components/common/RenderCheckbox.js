import React from "react";
import PropTypes from "prop-types";
import { Form, Checkbox } from "antd";

/**
 * @constant RenderCheckbox - Ant Design `Checkbox` component for redux-form.
 */

const propTypes = {
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  label: PropTypes.string.isRequired,
  meta: PropTypes.objectOf(PropTypes.any).isRequired,
};

const RenderCheckbox = (props) => (
  <Form.Item validateStatus={props.meta.touched ? props.meta.error && "error" : ""}>
    <Checkbox id={props.id} checked={props.input.value} {...props.input}>
      {props.label}
    </Checkbox>
  </Form.Item>
);

RenderCheckbox.propTypes = propTypes;

export default RenderCheckbox;
