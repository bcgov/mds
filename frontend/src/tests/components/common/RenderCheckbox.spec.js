import React from 'react';
import { shallow } from 'enzyme';
import RenderCheckbox from '@/components/common/RenderCheckbox';

let props = {};

const setupProps = () => {
  props = {
    id: 1,
    input: '',
    label: '',
    meta: {
      touched: false,
      error: false,
    },
  };
}

beforeEach(() => {
  setupProps();
});

describe('RenderCheckbox', () => {
  it('renders properly', () => {
    const wrapper = shallow(<RenderCheckbox {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
