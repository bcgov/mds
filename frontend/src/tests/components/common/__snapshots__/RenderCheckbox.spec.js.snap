// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderCheckbox renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RenderCheckbox
    id={1}
    input=""
    label=""
    meta={
      Object {
        "error": false,
        "touched": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Checkbox
        checked={undefined}
        id={1}
        indeterminate={false}
        prefixCls="ant-checkbox"
      >
        
      </Checkbox>,
      "colon": true,
      "hasFeedback": false,
      "prefixCls": "ant-form",
      "validateStatus": "",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "checked": undefined,
        "children": "",
        "id": 1,
        "indeterminate": false,
        "prefixCls": "ant-checkbox",
      },
      "ref": null,
      "rendered": "",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Checkbox
          checked={undefined}
          id={1}
          indeterminate={false}
          prefixCls="ant-checkbox"
        >
          
        </Checkbox>,
        "colon": true,
        "hasFeedback": false,
        "prefixCls": "ant-form",
        "validateStatus": "",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "checked": undefined,
          "children": "",
          "id": 1,
          "indeterminate": false,
          "prefixCls": "ant-checkbox",
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
